org.lflang/src/org/lflang/generator/ts/TSGenerator.kt

```diff
@@ -501,6 +502,22 @@ class TSGenerator(
         """.trimMargin()}
     }
 
+    /**
+     * Add necessary code to the source and necessary build supports to
+     * enable the requested serializations in 'enabledSerializations'
+     */
+    override fun enableSupportForSerialization(cancelIndicator: CancelIndicator?) {
+        for (serializer in enabledSerializers) {
+            when (serializer) {
+                SupportedSerializers.NATIVE -> {
+                    // No need to do anything at this point.
+                    println("Native serializer is enabled.")
+                }
+                else -> println("Serializer $serializer is enabled.")
```

Comment:
I'm not 100% sure what this method is meant to do, but I think that only for serialization methods that are indeed supported it should print "is enabled" and throw a `UnsupportedOperationException` otherwise. Maybe @Soroosh129 can comment on this. It looks like the implementation in `PythonGenerator` is similarly flawed.

Reply:
Right, this followed the implementation in `PythonGenerator`. By the way, TS federated execution fails without this fix, so I'd like to get this checked in to fix that...

Reply:
I agree with @lhstrh. It is better to throw an exception here if the `serializer` is not NATIVE. As best as I could remember, the solution in `PythonGenerator` was meant as a temporary measure because adding ROS 2 serialization was relatively straightforward. I haven't gotten around to that yet though.

Reply:
Thanks, Marten and Soroush. I made this thrown an exception in https://github.com/lf-lang/lingua-franca/pull/816/commits/1dc1ddb6b057a22be967be9f8c27f828bbb8e95c. Hope this makes more sense.